src/box.jl

```diff
@@ -2,17 +2,19 @@ export Box
 
 type Box{N,D,L} <: Shape{N}
     c::SVector{N,Float64} # box center
-    p::SMatrix{N,N,Float64,L} # projection matrix to box coordinates
     r::SVector{N,Float64}   # "radius" (semi-axis) in each direction
+    p::SMatrix{N,N,Float64,L} # projection matrix to box coordinates
     data::D             # auxiliary data
+    (::Type{Box{N,D,L}}){N,D,L}(c,r,p,data) = new{N,D,L}(c,r,p,data)
 end
 
-function Box(c::AbstractVector, d::AbstractVector,
-             axes=eye(length(c),length(c)), # columns are axes unit vectors
-             data=nothing)
-    (N = length(c)) == length(d) == size(axes,1) == size(axes,2) || throw(DimensionMismatch())
-    return Box{N,typeof(data),N^2}(c, inv(axes ./ sqrt.(sum(abs2,axes,1))), d*0.5, data)
-end
+Box{N,D,L,R<:Real}(c::SVector{N}, d::SVector{N},
+                   axes::SMatrix{N,N,R,L}=@SMatrix(eye(N)),  # columns are axes unit vectors
```

Comment:
Oh wow, that's very convenient!  I will make this change in the separate PR dedicated to all the changes for 0.6.